File: algo/125. Valid Palindrome.py
class Solution:
    def isPalindrome(self, s: 'str') -> 'bool':
        i = 0
        j = len(s)-1
        while i<j and i<len(s) and j>=0:
            while s[i].isalnum() != True:
                i += 1
                if i >= len(s):
                    return True
            while s[j].isalnum() != True:
                j -= 1
                if j < 0:
                    return True
            if s[i].lower() != s[j].lower():
                return False
            i += 1
            j -= 1
        return True

File: round3/125. Valid Palindrome.cs
public class Solution {
    public bool IsPalindrome(string s) {
        int left = 0, right = s.Length - 1;
        while (left < right) {
            if (!IsAlphaNumeric(s[left])) {
                left ++;
                continue;
            }
            if (!IsAlphaNumeric(s[right])) {
                right --;
                continue;
            }
            if (Char.ToLower(s[left]) != Char.ToLower(s[right])) return false;
            left ++;
            right --;
        }
        
        return true;
    }
    
    private bool IsAlphaNumeric(char c) {
        if ((c >= '0') && (c <= '9') || ((c >= 'a') && (c <='z')) || ((c >= 'A') && (c <= 'Z'))) return true;
        return false;
    }
}